refactor: migrating to discord.js to v14

pratik-GG opened the pull request:
I added temp v14.md file to keep track on progress

renovate[bot] committed:
fix(deps): update dependency discord.js to v14

pratik-GG committed:
fixed messageEmbed.addField deprecation warning

pratik-GG committed:
added accidentally deleted files

pratik-GG committed:
Merge pull request #1 from godofwarOP/kana

Fixed deprecation warnings

pratik-GG committed:
Merge branch 'Clytage:main' into main

pratik-GG committed:
upgraded config files to djs v14

pratik-GG committed:
upgraded utils to v14

pratik-GG committed:
upgraded general & developers commands to djs v14

pratik-GG committed:
upgraded events to djs v14
InteractionCreateEvent & ReadyEvent need to be fixed

pratik-GG committed:
upgraded structures to djs v14

pratik-GG committed:
Merge branch 'v14-test' into v14

pratik-GG committed:
added v14 roadmap for better progress tracking

pratik-GG committed:
upgraded moderation commands to djs v14
fixed help command

pratik-GG committed:
change activity type from 'PLAYING' to ActivityType.Playing

pratik-GG committed:
upgraded  &

pratik-GG committed:
upgraded  &

pratik-GG committed:
Changed typings/index.d.ts
MessageEmbed method is now EmbedBuilder

pratik-GG committed:
fixed method names and color names

pratik-GG committed:
fixed both handlers

pratik-GG committed:
Fixed ButtonPagination (only method names are changed)
filter need to be fixed

pratik-GG committed:
Merge branch 'v14' of github.com:godofwarOP/rawon into v14

fab1ofx reviewed (changes requested):
Actions failures in eslint or typescript, need a resolving.

`npm run lint && npm run tscompile` for check eslint or typescript issues

pratik-GG committed:
fixed commands

pratik-GG committed:
fixed config files
env.ts - changed exclude activity types from CUSTOM to ActivityType.Custom
index.ts - added MessageContent Intent to receive Privileged Message Content Intent data

pratik-GG committed:
notfixed : readyevent
the setPresence method is not working

pratik-GG committed:
change exclude activity type from CUSTOM to ActivityType.Custom

pratik-GG committed:
fixed utilities

pratik-GG committed:
updated v14 roadmap

Mednoob commented on src/commands/developers/EvalCommand.ts:
```diff
@@ -56,8 +56,7 @@ export class EvalCommand extends BaseCommand {
             const error = isTooLong ? `${await this.hastebin(cleaned)}.js` : `\`\`\`js\n${cleaned}\`\`\``;
-
-            embed.setColor("RED").addFields([{ name: i18n.__("commands.developers.eval.errorString"), value: error }]);
+            embed.setColor("Red").addFields([{ name: i18n.__("commands.developers.eval.errorString"), value: error }]);
```
The blank line that was here shouldn't be deleted. Variable declaration should be separated from function/method calls.

Mednoob commented on src/commands/moderation/KickCommand.ts:
```diff
@@ -62,13 +62,18 @@ export class KickCommand extends BaseCommand {
+                        .addFields([
+                            {
+                                name: i18n.__("commands.moderation.common.reasonString"),
+                                value: reason.toString()
```
`.toString()` is not required since `reason` is already a string

Mednoob commented on src/commands/moderation/WarnCommand.ts:
```diff
@@ -56,16 +56,16 @@ export class WarnCommand extends BaseCommand {
                 {
                     name: i18n.__("commands.moderation.common.reasonString"),
-                    value: displayReason
+                    value: displayReason.toString()
```
`displayReason` is already a string

Mednoob commented on src/events/InteractionCreateEvent.ts:
```diff
@@ -56,10 +63,10 @@ export class InteractionCreateEvent extends BaseEvent {
             const data = interaction.options.getUser("user") ?? interaction.options.getMessage("message");
             const cmd = this.client.commands.find(x =>
-                (data as { type: string }).type === "MESSAGE"
+                data?.toJSON() === ApplicationCommandType.Message
```
Is it just me or is it missing something like `.type` after `.toJSON()`?

Mednoob commented on src/events/MessageCreateEvent.ts:
```diff
@@ -2,19 +2,25 @@ import { createEmbed } from "../utils/functions/createEmbed";
+                message.channel.type === ChannelType.DM
+                    ? ChannelType.DM.toString()
+                    : `${message.channel.name}(${message.channel.id})`
+            ],
```
You don't need to change the `DM` string with enum value on the ternary true statement

Mednoob commented on src/events/MessageCreateEvent.ts:
```diff
@@ -2,19 +2,25 @@ import { createEmbed } from "../utils/functions/createEmbed";
 
-        if (message.author.bot || message.channel.type === "DM" || !this.client.commands.isReady) return message;
+        if (message.author.bot || message.channel.type === ChannelType.DM || !this.client.commands.isReady)
+            return message;
```
Use curly bracket if you want to do multi-line if statement

Mednoob commented on src/structures/CommandContext.ts:
```diff
@@ -137,8 +143,13 @@ export class CommandContext {
+        OLD isSelectMenu method is now deprecated
+        NEW use isStringSelectMenu method  
+        ref :- https://discord.js.org/#/docs/discord.js/main/class/CommandInteraction?scrollTo=isSelectMenu
+        */
```
<img src="https://media.tenor.com/XPzoXNYIuy8AAAAC/oozora-subaru-hololive.gif" width="200"/>

You can remove the comment.

Mednoob commented on src/utils/structures/ClientUtils.ts:
```diff
@@ -46,6 +46,11 @@ export class ClientUtils {
+         * This method is throwing reference error
+         * ChannelType is not defined, even though it is being imported
+         * for now i added channel types directly instead of using enums
+         */
```
This needs to be solved, ofc

pratik-GG replied:
this error is so confusing. I imported ChannelType from discord.js correctly but still it throws reference error
and because of this error the setPresence method is not working.
what i did now is instead of using `ChannelType.GuildText` i used `0` and it's working.

Mednoob replied:
Finally found the cause of the error. Apparently, I just remembered that `broadcastEval` stringifies the function to be evaluated by shards. But since there's no `ChannelType` variable in the environment where the function is being evaluated, the code emits an error saying that `ChannelType` is not defined.

We can fix this by adding `ChannelType` to the context (see second argument of `broadcastEval`):
```javascript
{ context: { textOnly, voiceOnly, types: ChannelType } }
```

In the function, there's already `t` variable as the second parameter. That's the context. So, in the function, to access the channel types you can access `types` property from the context:
```javascript
if (t.textOnly) {
    return ch.type === t.types.GuildText ||
        ch.type === t.types.PublicThread ||
        ch.type === t.types.PrivateThread;
} else if (t.voiceOnly) {
    return ch.type === t.types.GuildVoice;
}
```

pratik-GG replied:
wow, i was working on same file and found same fix :joy:

Mednoob replied:
wait, really? wow 🤣

pratik-GG committed:
Fix:
EvalCommands: added extra space above embed#setColor
Kick & WarnCommand: removed #toString() from reason

pratik-GG committed:
Fix: Events
InteractionEventCreate: #type property is not available on options#getUser method, so i added an extra if check to see if data is Message
MessageCreateEvent: fixed DM value & i hate prettier XD

pratik-GG committed:
removed comments

renovate[bot] committed:
fix(deps): update dependency discord.js to v14

pratik-GG committed:
Merge branch 'Clytage:main' into v14

pratik-GG committed:
Fix: ClientUtils

pratik-GG committed:
Fix: setPresence method
PLAYING.. etc are not valid ActivityType we need ID of that activity
1. Added new type EnvActivityTypes
2. toCapitalCase method to convert PLAYING to Playing
3. search for id in ActivityType using ActivityType['Playing'] which returns 0

pratik-GG committed:
Merge branch 'renovate/discord.js-14.x' into v14

pratik-GG committed:
updated roadmap

pratik-GG committed:
Update v14.md

renovate[bot] committed:
fix(deps): update dependency discord.js to v14

pratik-GG committed:
Merge branch 'renovate/discord.js-14.x' into v14

renovate[bot] committed:
fix(deps): update dependency discord.js to v14

pratik-GG committed:
Merge branch 'renovate/discord.js-14.x' into v14

pratik-GG committed:
fixed a minor typo

pratik-GG committed:
Fix typo in UnBanCommand
Fix SetMuteCommand

pratik-GG committed:
Fix
not sending modlogs for unban event, the fetch method always returns undefined

pratik-GG committed:
fixed conflicts caused by renovate bot

pratik-GG committed:
updated roadmap

pratik-GG committed:
Merge branch 'v14' of github.com:godofwarOP/rawon into v14

pratik-GG committed:
fixed minor typo in en.json

pratik-GG committed:
Merge branch 'Clytage:main' into v14

pratik-GG committed:
fixed @typescript-eslint/no-unnecessary-type-assertion error

Mednoob commented on lang/en.json:
```diff
@@ -146,7 +146,7 @@
                 "slashMemberDescription": "Who do you like to un-ban?",
                 "slashReasonDescription": "Un-ban reason",
-                "ubanSuccess": "**{user}** has been **`UN-BANNED`** from the server.",
+                "unbanSuccess": "**{user}** has been **`UN-BANNED`** from the server.",
```
Change the key in other localization files as well.

pratik-GG replied:
:+1: sorry i was distracted didn't saw other files

pratik-GG committed:
fixed typo ubanSuccess to unbanSuccess

Mednoob reviewed (approved):
LGTM

Commit:
Merge branch 'renovate/discord.js-14.x' into v14
